mu-plugins/central-hub/tests/phpunit/integration/meta-data/saveMetaBoxes.php

```diff
@@ -0,0 +1,322 @@
+use function add_post_meta;
+use function get_post_meta;
+use function delete_post_meta;
+use spiralWebDb\Cornerstone\Tests\Integration\Test_Case;
+
+/**
+ * Class Tests_SaveMetaBoxes
+ *
+ * @package spiralWebDb\centralHub\Tests\Integration\Metadata
+ * @group   meta-data
+ */
+class Tests_SaveMetaBoxes extends Test_Case {
+
+	/**
+	 * Instance of the post object for each test.
+	 *
+	 * @var WP_Post
+	 */
+	protected $post;
+
+	/**
+	 * Empty the store before starting these tests.
+	 */
+	public static function setUpBeforeClass() {
+		self::empty_the_store();
+	}
+
+	/**
+	 * Set up each test.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		// Create and get the post_id via the factory method.
+		$this->post = self::factory()->post->create();
+	}
+
+	/**
+	 * Cleans up the test environment after each test.
+	 */
+	public function tearDown() {
+		parent::tearDown();
+		// Clean up.
+		self::empty_the_store();
+		$_POST = [];
+		delete_post_meta( $this->post, 'role' );
+		delete_post_meta( $this->post, 'residence_city' );
+		delete_post_meta( $this->post, 'residence_state' );
+		delete_post_meta( $this->post, 'tour_number' );
+	}
+
+	/*
+	* Test save_meta_boxes() should register to add_action( save_post' ) when event fires.
+	*/
+	public function test_function_should_register_to_action_hook_when_event_fires() {
+		$this->assertTrue( has_action( 'save_post' ) );
+		// `has_action()` returns the priority level of the callback registered to the event ($tag).
+		$this->assertEquals( 10, has_action( 'save_post', 'spiralWebDB\Metadata\save_meta_boxes' ) );
+	}
+
+	/**
+	 * Test save_meta_boxes() should not save when no meta box keys are in the Config Store.
+	 */
+	public function test_should_not_save_when_no_meta_box_keys_are_in_store() {
+		$configs = [];
+		foreach ( $configs as $store_key => $config_to_store ) {
+			loadConfig( $store_key, $config_to_store );
+		}
+		$_POST = [
+			'post_ID'            => $this->post,
+			'post_status'        => 'publish',
+			'members'            => [],
+			'members_nonce_name' => wp_create_nonce( 'members_nonce_action' ),
+		];
+
+		$this->assertNull( save_meta_boxes( $this->post ) );
+	}
+
+	/**
+	 * Test save_meta_boxes() should not save when the store_key does not start with `meta_box.`.
+	 */
+	public function test_should_not_save_when_store_key_does_not_start_with_meta_box() {
+		$configs = [
+			'notametabox.members' => [
+				'custom_fields' => [
+					'role'            => 'Soprano',
+					'residence_city'  => 'Chicago',
+					'residence_state' => 'IL',
+					'tour_number'     => '3',
+				],
+			],
+		];
+		foreach ( $configs as $store_key => $config_to_store ) {
+			loadConfig( $store_key, $config_to_store );
+		}
+		$_POST = [
+			'post_ID'            => $this->post,
+			'post_status'        => 'publish',
+			'members'            => [],
+			'members_nonce_name' => wp_create_nonce( 'members_nonce_action' ),
+		];
+
+		$this->assertNull( save_meta_boxes( $this->post ) );
```

Comment:
@rgadon107 Like above, this test doesn't validate anything.

In this one, you are attempting to test that it does not save (or update) when the configs exist but there's nothing to save in the `$_POST`.

Let's think about how to test it. We want to validate that it didn't save. There are a couple of ways to double check that:

1. None of the configured post meta exists in the db _before_ running `save_meta_boxes()`.
2. There are values in the db for the configured post meta; but they don't change after running `save_meta_boxes()`.

 Notice that either method tests that a save or update does not occur.

What do you think?

Reply:
I edited the second integration test to test _both_ use cases: 

1) None of the configured post meta exists in the db before running `save_meta_boxes()`. DB is not updated.
2) There are values in the DB for the configured post meta. Those values don't update after running `save_meta_boxes()`.

Integration test passed, and changed committed.